### PR TITLE
Styx docker environment: add NGINX origins

### PR DESCRIPTION
**Pull request by mikkokar**
Replaces Apache httpd with NGINX based origins.

### CODE REVIEW

File: system-tests/docker-test-env/docker-compose.yml [hunk 1]
```diff
@@ -34,9 +35,6 @@ services:
     volumes:
       - ./styx-config:/styx/config
       - ./styx-logs:/styx/logs
-    links:
```
Comment (dvlato) [inline system-tests/docker-test-env/docker-compose.yml]: About time :)

File: system-tests/docker-test-env/docker-compose.yml [hunk 2]
```diff
@@ -1,17 +1,18 @@
 version: '3'
 services:
-
-  httpd-01:
-    image: httpd:2.4
+  origin:
+    image: nginx:stable-alpine
     ports:
-      - "8081:80"
+      - "9091:8080"
```
Comment (dvlato) [inline system-tests/docker-test-env/docker-compose.yml]: I would rather start the nginx listener in the same ports we are going to expose externally.

File: system-tests/docker-test-env/nginx/testCredentials.crt [hunk 3]
```diff
@@ -0,0 +1,22 @@
+-----BEGIN CERTIFICATE-----
```
Comment (dvlato) [inline system-tests/docker-test-env/nginx/testCredentials.crt]: This certification expired on 2015.
Why are we declaring a HTTPS port for Nginx if are not using it in our origin configuration?

File: system-tests/docker-test-env/docker-compose.yml (hunk 2 above)
Comment (mikkokar) [reply system-tests/docker-test-env/docker-compose.yml]: Hi @dvlato 

I'd like to keep all http services behind the same port (8080). It is easier to configure all of styx, toxi-proxy, and nginx etc when the port is always the same. The ports won't conflict inside docker network: each service will have its own host name.

Therefore the ports are exposed only for the debugging purposes, and on the host hey need to be unique.

Therefore I'd keep it as it is.

File: system-tests/docker-test-env/nginx/testCredentials.crt (hunk 3 above)
Comment (mikkokar) [reply system-tests/docker-test-env/nginx/testCredentials.crt]: For now this only servers as a configuration example for enabling TLS on NGINX. I couldn't get this to work without a certificate, and this old, self-signed test certificate appears to do just fine.

I could add this to the origins configuration if that helps?

Comment (mikkokar) [reply system-tests/docker-test-env/nginx/testCredentials.crt]: I have now added a secure origin.

Comment (dvlato) [reply system-tests/docker-test-env/nginx/testCredentials.crt]: Maybe we should add toxyproxy config for that secure origin as well.

**Pull request merged**